Automattic/jetpack
# Instant Search: Restore caching to API requests

Take 2 of #13714.

#### Changes proposed in this Pull Request:
Re-implements caching of API requests using [`fast-json-stable-stringify`](https://bundlephobia.com/result?p=fast-json-stable-stringify@2.0.0) instead of [`object-hash`](https://bundlephobia.com/result?p=object-hash@2.0.0). 

`fast-json-stable-stringify` is a significantly smaller library that provides what we're looking for: a way to deterministically hash an array of input arguments.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
No.

#### Testing instructions:
1. Setup Instant Search as per instructions in the [readme](https://github.com/Automattic/jetpack/blob/instant-search-master/modules/search/instant-search/README.md#testing-instructions).
2. Enter a query and ensure the results render as expected.
3. Enter a different query and ensure the results render as expected.
4. Enter the same query from step 2 and ensure the results render near-instantaneously. You can also check that no new API request has been fired off in the network tab of your developer tools.

#### Proposed changelog entry for your changes:
None.
